src/Aspire.Hosting.Analyzers/Infrastructure/BoundedCacheWithFactory.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.Analyzers.Infrastructure;
+
+// This type is copied from https://github.com/dotnet/roslyn-analyzers/blob/9b58ec3ad33353d1a523cda8c4be38eaefc80ad8/src/Utilities/Compiler/BoundedCacheWithFactory.cs
```

Comment:
This is unfortunate that it needs to be copied to everywhere that wants to be an analyzer.

Reply:
Yeah. I was talking about another case where we wanted to share sources earlier today and day-dreamed about built-in support for producing source packages directly from the CLI, e.g. `dotnet pack --id MySourcePackage --version 0.0.1 --description "My Great source package" --sources ./**/*.cs,./**/*.resx`, so we could avoid having to create special projects everywhere we want to share sources.

Reply:
See also https://github.com/dotnet/aspire/tree/main/src/Tools/ConfigurationSchemaGenerator/RuntimeSource where we copied a TON of the Configuration.Binder source generator code. 😞